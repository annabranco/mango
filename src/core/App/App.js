import React from "react";
import { BrowserRouter, Navigate, Route, Routes } from "react-router-dom";
import SETUP from "../../config/setup.js";
import { useStateWithLabel } from "../../utils/hooks";
import Home from "../../views/Home/Home";
import RangeView from "../../views/RangeView/RangeView";
import { RANGE } from "../../constants";
import { GlobalStyles } from "../../globals.styles";
import { MainArea } from "./App.styles";

const App = () => {
  const [currentMaxValue, changeCurrentMaxValue] = useStateWithLabel(
    undefined,
    "currentMaxValue"
  );
  const [currentMinValue, changeCurrentMinValue] = useStateWithLabel(
    undefined,
    "currentMinValue"
  );
  const [currentValue, changeCurrentValue] = useStateWithLabel(
    undefined,
    "currentValue"
  );
  const [rangeType, changeRangeType] = useStateWithLabel(RANGE, "rangeType");

  return (
    <>
      <GlobalStyles />
      <MainArea>
        <BrowserRouter>
          <Routes>
            <Route
              exact
              path="/exercise1"
              element={
                <RangeView
                  changeCurrentMaxValue={changeCurrentMaxValue}
                  changeCurrentMinValue={changeCurrentMinValue}
                  currentMaxValue={currentMaxValue}
                  currentMinValue={currentMinValue}
                  currentValue={currentValue}
                  displayMarks={false}
                  onChange={changeCurrentValue}
                  type={rangeType}
                  unit={SETUP.numberUnit}
                />
              }
            />
            <Route
              exact
              path="/exercise2"
              element={
                <RangeView
                  changeCurrentMaxValue={changeCurrentMaxValue}
                  changeCurrentMinValue={changeCurrentMinValue}
                  currentMaxValue={currentMaxValue}
                  currentMinValue={currentMinValue}
                  currentValue={currentValue}
                  displayMarks={true}
                  onChange={changeCurrentValue}
                  type={rangeType}
                  unit={SETUP.numberUnit}
                />
              }
            />
            <Route index element={<Home />} />
            <Route path="*" element={<Navigate to="/" />} />
          </Routes>
        </BrowserRouter>
      </MainArea>
    </>
  );
};

export default App;
