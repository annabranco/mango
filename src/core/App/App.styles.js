import styled from "styled-components";

export const MainArea = styled.main`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 50px;
`;
MainArea.displayName = "MainArea";

export const SelectionText = styled.p`
  margin: 20px auto 40px;
  margin-left: 15vw;
  width: 100%;
  font-size: 1.4rem;
  text-align: left;
`;
SelectionText.displayName = "SelectionText";
