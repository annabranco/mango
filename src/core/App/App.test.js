import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import toJson from "enzyme-to-json";

import App from "./App";

configure({ adapter: new Adapter() });

describe("[App] component with normal values", () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<App />);
  });

  it("should match the snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
