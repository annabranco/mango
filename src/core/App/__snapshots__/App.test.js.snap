// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[App] component with normal values should match the snapshot 1`] = `
<Fragment>
  <Memo(l) />
  <MainArea>
    <BrowserRouter>
      <Routes>
        <Route
          element={
            <RangeView
              changeCurrentMaxValue={[Function]}
              changeCurrentMinValue={[Function]}
              currentMaxValue={100}
              currentMinValue={1}
              currentValue={0}
              displayMarks={false}
              onChange={[Function]}
              type="SINGLE"
              unit="€"
            />
          }
          exact={true}
          path="/exercise1"
        />
        <Route
          element={
            <RangeView
              changeCurrentMaxValue={[Function]}
              changeCurrentMinValue={[Function]}
              currentMaxValue={100}
              currentMinValue={1}
              currentValue={0}
              displayMarks={true}
              onChange={[Function]}
              type="SINGLE"
              unit="€"
            />
          }
          exact={true}
          path="/exercise2"
        />
        <Route
          element={<Home />}
          index={true}
        />
        <Route
          element={
            <Navigate
              to="/"
            />
          }
          path="*"
        />
      </Routes>
    </BrowserRouter>
  </MainArea>
</Fragment>
`;
